src/main/java/com/winner_is_kungen/tda367/model/Blueprint.java

```diff
@@ -29,6 +30,41 @@
 	 */
 	private final EventBus eventBus = new EventBus(eventConnection, eventComponent);
 
+
+	/**
+	 * Gets list of components to write to file
+	 *
+	 * @return List of component data for writing
+	 */
+
+	public ArrayList<String> getComponentsForWrite() {
```

Comment:
This should be handled in the `WriteFile` service. The model should not have to care about file formats or even files.